src/rgw/rgw_rados.cc

```diff
@@ -13017,7 +13017,7 @@ int RGWRados::cls_bucket_head(const RGWBucketInfo& bucket_info, int shard_id, ma
 
   map<int, struct rgw_cls_list_ret>::iterator iter = list_results.begin();
   for(; iter != list_results.end(); ++iter) {
-    headers[oids[iter->first]] = iter->second.dir.header;
+    headers[iter->first] = iter->second.dir.header;
```

Comment:
since `list_results` is a local variable, we can use `std::move(iter->second.dir.header)` here instead